vxtelegram/telegram.py

```diff
@@ -157,31 +185,104 @@ def handle_outbound_message(self, message):
         url = self.get_outbound_url('sendMessage')
         http_client = HTTPClient(self.agent_factory())
 
-        try:
-            r = yield http_client.post(
-                url=url,
-                data=json.dumps(outbound_msg),
-                headers={'Content-Type': ['application/json']},
-            )
+        r = yield http_client.post(
+            url=url,
+            data=json.dumps(outbound_msg),
+            headers={'Content-Type': ['application/json']},
+            allow_redirects=False,
+        )
+
         # Telegram redirects our request if our bot token is invalid
-        except ResponseFailed as e:
-            yield self.outbound_failure(message_id,
-                                        'Invalid token (redirected)\n%s' % e)
+        if r.code == http.FOUND:
+            yield self.outbound_failure(
+                message_id=message_id,
+                message='Invalid bot token (redirected)',
+                error='Redirected',
+            )
             return
+
         try:
             res = yield r.json()
-        except ValueError as e:
-            yield self.outbound_failure(message_id,
-                                        'Expected JSON response\n%s' % e)
+        except ValueError:
+            yield self.outbound_failure(
+                message_id=message_id,
+                message='Expected JSON response',
+                error='error code %s' % r.code,
```

Comment:
I think it would be helpful if the status included the code itself in `details` (something like `{..., details: {code: '200', ...}}`), so that one could use the code programmatically without needing to parse the code out of a human-readable message. I also think we might want the response body for this case, since it might be helpful for debugging why the transport is down.

Maybe we should change `outbound_failure` to take in a `description` dictionary instead?